src/server/pfs/server/tracing/wrapper.go

```diff
@@ -0,0 +1,261 @@
+	opentracing.endTK()
+	return s.APIClient.CopyFile(context.Context, *CopyFileRequest)
+}
+
+// GetFile implements the corresponding method of the pfs.APIServer interface
+func (s *tracingServer) GetFile(*GetFileRequest, API_GetFileServer) error {
+	opentracing.start_TK
+	opentracing.endTK()
+	return s.APIClient.GetFile(*GetFileRequest, API_GetFileServer)
+}
+
+// InspectFile implements the corresponding method of the pfs.APIServer interface
+func (s *tracingServer) InspectFile(context.Context, *InspectFileRequest) (*FileInfo, error) {
+	opentracing.start_TK
+	opentracing.endTK()
+	return s.APIClient.InspectFile(context.Context, *InspectFileRequest)
+}
+
+// ListFile implements the corresponding method of the pfs.APIServer interface
+func (s *tracingServer) ListFile(context.Context, *ListFileRequest) (*FileInfos, error) {
+	opentracing.start_TK
+	opentracing.endTK()
+	return s.APIClient.ListFile(context.Context, *ListFileRequest)
+}
+
+// ListFileStream implements the corresponding method of the pfs.APIServer interface
+func (s *tracingServer) ListFileStream(*ListFileRequest, API_ListFileStreamServer) error {
+	opentracing.start_TK
+	opentracing.endTK()
+	return s.APIClient.ListFileStream(*ListFileRequest, API_ListFileStreamServer)
+}
+
+// WalkFile implements the corresponding method of the pfs.APIServer interface
+func (s *tracingServer) WalkFile(*WalkFileRequest, API_WalkFileServer) error {
+	opentracing.start_TK
+	opentracing.endTK()
+	return s.APIClient.WalkFile(*WalkFileRequest, API_WalkFileServer)
+}
+
+// GlobFile implements the corresponding method of the pfs.APIServer interface
+func (s *tracingServer) GlobFile(context.Context, *GlobFileRequest) (*FileInfos, error) {
+	opentracing.start_TK
+	opentracing.endTK()
+	return s.APIClient.GlobFile(context.Context, *GlobFileRequest)
+}
+
+// GlobFileStream implements the corresponding method of the pfs.APIServer interface
+func (s *tracingServer) GlobFileStream(*GlobFileRequest, API_GlobFileStreamServer) error {
+	opentracing.start_TK
+	opentracing.endTK()
+	return s.APIClient.GlobFileStream(*GlobFileRequest, API_GlobFileStreamServer)
+}
+
+// DiffFile implements the corresponding method of the pfs.APIServer interface
+func (s *tracingServer) DiffFile(context.Context, *DiffFileRequest) (*DiffFileResponse, error) {
+	opentracing.start_TK
+	opentracing.endTK()
+	return s.APIClient.DiffFile(context.Context, *DiffFileRequest)
+}
+
+// DeleteFile implements the corresponding method of the pfs.APIServer interface
+func (s *tracingServer) DeleteFile(context.Context, *DeleteFileRequest) (*types.Empty, error) {
+	opentracing.start_TK
+	opentracing.endTK()
+	return s.APIClient.DeleteFile(context.Context, *DeleteFileRequest)
+}
+
+// DeleteAll implements the corresponding method of the pfs.APIServer interface
+func (s *tracingServer) DeleteAll(context.Context, *types.Empty) (*types.Empty, error) {
+	opentracing.start_TK
+	opentracing.endTK()
+	return s.APIClient.DeleteAll(context.Context, *types.Empty)
+}
```

Comment:
Would it possibly make sense to embed these tracing calls in the `Log` calls that happen at the beginning of every request? I'm a little worried about the amount of boilerplate we're going to be requiring people to write when they add a new method to one of the servers.

Reply:
This file's gone, though I'm trying to move some of this boilerplate into interceptors

Reply:
Update: An issue I ran into moving boilerplate into interceptors had to do with PutFile: I would need to return a custom ClientStream/ServerStream to tag the stream with the files being Put. While I ultimately want to do that, I think it might make sense to break that out into a separate project

Reply:
Cool, I wouldn't  do the more extended stuff for PutFile yet, makes sense as a separate project.